mxcubecore/HardwareObjects/BeamlineActions.py

```diff
@@ -185,23 +186,27 @@ def _get_command_object_class(self, path_str):
         return _cls
 
     def init(self):
-        command_list = ast.literal_eval(
-            self.get_property("commands").strip().replace("\n", "")
-        )
+        command_list = self.get_property("commands")
+        if isinstance(command_list, str):
+            command_list = ast.literal_eval(command_list.strip().replace("\n", ""))
 
         for command in command_list:
-            attrname = camel_to_snake(command["command"].split(".")[-1])
+            action = command["command"].split(".")[-1]
+            attrname = camel_to_snake(action)
+            if command.get("disabled", False):
+                self.log.warning(f"Action {attrname} is disabled in the configuration.")
+                continue
 
             if hasattr(self, attrname):
-                msg = (
-                    "Command with name %s already exists"
-                    % command["command"].split(".")[-1]
-                )
+                msg = f"Action {action} already exists"
                 self.log.warning(msg)
                 continue
 
             if command["type"] == "annotated":
-                _cls = self._get_command_object_class(command["command"])
+                try:
+                    _cls = self._get_command_object_class(command["command"])
+                except:
+                    self.log.exception(f"failed to load annotated action {action}")
                 fname = camel_to_snake(_cls.__name__)
                 _cls_inst = _cls(self, fname.replace("_", " ").title(), fname)
```

Comment:
if `_cls = self._get_command_object_class(command["command"])` would fail, this line will fail again as `_cls` won't be defined 

```suggestion
                else:
                    fname = camel_to_snake(_cls.__name__)
                    _cls_inst = _cls(self, fname.replace("_", " ").title(), fname)
```

Reply:
nice catch @dominikatrojanowska I'll merge this tomorrow as soon as I test it.